pages/_app.js

```diff
@@ -2,17 +2,142 @@ import GlobalStyle from "../styles";
+
+    const formData = new FormData(event.target);
+    const newContact = Object.fromEntries(formData);
+
+    const newContactId = uid();
+
+    const date = new Date();
+    const currentUtcDateTime = date.toISOString();
+
+    const formattedContact = {
+      id: newContactId,
+      firstName: newContact.firstName,
+      middleName: newContact.middleName,
+      lastName: newContact.lastName,
+      nickName: newContact.nickName,
+      gender: newContact.gender,
+      profilePicture: "",
+      dateOfBirth: newContact.dateOfBirth,
+      deceased: false,
+      email: [],
+      phone: [],
+      address: [],
+      socialMedia: [],
+      notes: newContact.notes,
+      dateCreated: currentUtcDateTime,
+      dateDeleted: "",
+      isSampleData: false,
+    };
+
+    if (newContact.deceased) {
+      formattedContact.deceased = true;
+    }
+
+    if (newContact.emailOne) {
+      formattedContact.email.push({
+        value: newContact.emailOne,
+        type: newContact.emailOneType,
+      });
+    }
+    if (newContact.emailTwo) {
+      formattedContact.email.push({
+        value: newContact.emailTwo,
+        type: newContact.emailTwoType,
+      });
+    }
+
+    if (newContact.phoneOne) {
+      formattedContact.phone.push({
+        value: newContact.phoneOne,
+        type: newContact.phoneOneType,
+      });
+    }
+
+    if (newContact.phoneTwo) {
+      formattedContact.phone.push({
+        value: newContact.phoneTwo,
+        type: newContact.phoneTwoType,
+      });
+    }
+
+    if (
+      newContact.addressOneType ||
+      newContact.addressOneStreet ||
+      newContact.addressOneZipCode ||
+      newContact.addressOneCity ||
+      newContact.addressOneCountry
+    ) {
+      formattedContact.address.push({
+        type: newContact.addressOneType,
+        street: newContact.addressOneStreet,
+        zipCode: newContact.addressOneZipCode,
+        city: newContact.addressOneCity,
+        country: newContact.addressOneCountry,
+      });
+    }
+
+    if (
+      newContact.addressTwoType ||
+      newContact.addressTwoStreet ||
+      newContact.addressTwoZipCode ||
+      newContact.addressTwoCity ||
+      newContact.addressTwoCountry
+    ) {
+      formattedContact.address.push({
+        type: newContact.addressTwoType,
+        street: newContact.addressTwoStreet,
+        zipCode: newContact.addressTwoZipCode,
+        city: newContact.addressTwoCity,
+        country: newContact.addressTwoCountry,
+      });
+    }
+
+    if (newContact.instagram) {
+      formattedContact.socialMedia.push({
+        platform: "Instagram",
+        username: newContact.instagram,
+      });
+    }
+    if (newContact.twitter) {
+      formattedContact.socialMedia.push({
+        platform: "Twitter",
+        username: newContact.twitter,
+      });
+    }
+    if (newContact.facebook) {
+      formattedContact.socialMedia.push({
+        platform: "Facebook",
+        username: newContact.facebook,
+      });
+    }
```

Comment:
all there if cases and pushing into an object, that you just created, seems a little too much to me. Try to find a shorter way of writing, check operators like `ternery`, `&&` or `||` to make your code better understandable. 

and: this sould be done in the Form component, not in app.js ;)

Reply:
@julialoeschel 

GPT made it easier for me using .filter at the end of the array. For example: 

```
      email: [
        {
          value: newContact.emailOne,
          type: newContact.emailOneType,
        },
        {
          value: newContact.emailTwo,
          type: newContact.emailTwoType,
        },
      ].filter((email) => email.value),
      phone: [
        {
          value: newContact.phoneOne,
          type: newContact.phoneOneType,
        },
        {
          value: newContact.phoneTwo,
          type: newContact.phoneTwoType,
        },
      ].filter((phone) => phone.value),
      address: ...
```

How can I do the same for socialMedia array?
```
    if (newContact.instagram) {
      formattedContact.socialMedia.push({
        platform: "Instagram",
        username: newContact.instagram,
      });
    }
    if (newContact.twitter) {
      formattedContact.socialMedia.push({
        platform: "Twitter",
        username: newContact.twitter,
      });
    }
    if (newContact.facebook) {
      formattedContact.socialMedia.push({
        platform: "Facebook",
        username: newContact.facebook,
      });
    }
```

I tried with a ternary but how do I not return nothing when the condition is falsy? 
```
      socialMedia: [
        newContact.instagram
          ? {
              platform: "Instagram",
              username: newContact.instagram,
            }
          : null,
      ],
```
Trying the above I get NULL into my array which leads to an error when displaying my data later on.
![Screenshot 2023-07-13 at 16 45 23](https://github.com/josch87/Gunther/assets/33511915/18c9a665-1f8c-4905-838c-ff22d8b557ca)

Reply:
After speaking with Jessica and Dominik it works just the same:

```
      socialMedia: [
        {
          platform: "Instagram",
          username: newContact.instagram,
        },
        {
          platform: "Twitter",
          username: newContact.twitter,
        },
        {
          platform: "Facebook",
          username: newContact.facebook,
        },
      ].filter((socialMedia) => socialMedia.username),
```

Reply:
```
.filter((socialMedia) => socialMedia.username)
```
is like saying
```
.filter((socialMedia) => (socialMedia.username ? true : false))
```